CHANGELOG/CHANGELOG-3.4.md

```diff
@@ -6,6 +6,9 @@ Previous change logs can be found at [CHANGELOG-3.3](https://github.com/etcd-io/
 
 ## v3.4.31 (TBD)
 
+### Dependencies
+- Compile binaries using [go 1.21.6](https://github.com/etcd-io/etcd/pull/17368)
```

Comment:
Minor comment: Usually we put section "Dependencies" at the bottom of the each patch.